### PR TITLE
dev-util/ftjam: EAPI=7 bump

### PR DESCRIPTION
**Pull request by mm1ke**
Hi,

I've updated dev-util/ftjam for EAPI=7 and improved the ebuild a bit.
Please review.

Closes: https://bugs.gentoo.org/686616
Signed-off-by: Michael Mair-Keimberger <m.mairkeimberger@gmail.com>

```diff
--- ftjam-2.5.3_rc2-r1.ebuild   2019-05-21 18:44:19.498014043 +0200
+++ ftjam-2.5.3_rc2-r2.ebuild   2019-05-23 19:24:01.846189294 +0200
@@ -1,44 +1,34 @@
 # Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=0
+EAPI=7
 
-inherit eutils toolchain-funcs versionator
+inherit toolchain-funcs
 
-MY_PV=$(delete_version_separator _)
+MY_PV=$(ver_rs 3 "")
 
-DESCRIPTION="FTJam is a 100% compatible, enhanced implementation of the make alternative Jam"
+DESCRIPTION="A 100% compatible, enhanced implementation of the make alternative Jam"
 HOMEPAGE="http://freetype.sourceforge.net/jam/index.html"
 SRC_URI="http://david.freetype.org/jam/ftjam-${MY_PV}.tar.bz2"
 
 LICENSE="perforce GPL-2"
 SLOT="0"
-KEYWORDS="amd64 arm64 hppa ppc ppc64 sparc x86 ~amd64-linux ~x86-linux ~ppc-macos"
-IUSE=""
+KEYWORDS="~amd64 ~arm64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
 
-DEPEND="!dev-util/jam
-       sys-devel/bison"
-RDEPEND="!dev-util/jam"
+DEPEND="sys-devel/bison"
 
 S=${WORKDIR}/${PN}-${MY_PV}
 
-src_unpack() {
-       unpack ${A}
-       cd "${S}"
-       epatch "${FILESDIR}"/${PN}-2.5.3-nostrip.patch
-       epatch "${FILESDIR}"/${PN}-2.5.3-i-hate-yacc.patch
-       epatch "${FILESDIR}"/${PN}-2.5.3-false-flags.patch
-#      epatch "${FILESDIR}"/${PN}-2.5.3-debug-commandline.patch  # development only
-}
+DOCS=( README README.ORG CHANGES INSTALL RELNOTES )
+HTML_DOCS=( Jam.html Jambase.html Jamfile.html )
 
-src_compile() {
-       tc-export CC
-       econf
-       emake || die "emake failed"
-}
+PATCHES=(
+       "${FILESDIR}"/${PN}-2.5.3-nostrip.patch
+       "${FILESDIR}"/${PN}-2.5.3-i-hate-yacc.patch
+       "${FILESDIR}"/${PN}-2.5.3-false-flags.patch
+)
 
-src_install() {
-       emake DESTDIR="${D}" install || die
-       dodoc README README.ORG CHANGES INSTALL RELNOTES
-       dohtml Jam.html Jambase.html Jamfile.html
+src_prepare() {
+       default
+       tc-export CC
 }
```

### CODE REVIEW

File: dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+DESCRIPTION="A 100% compatible, enhanced implementation of the make alternative Jam"
+HOMEPAGE="http://freetype.sourceforge.net/jam/index.html"
+SRC_URI="http://david.freetype.org/jam/ftjam-${MY_PV}.tar.bz2"
+
+LICENSE="perforce GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~hppa ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos"
+
+DEPEND="sys-devel/bison"
```
Comment (a17r) [inline dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild]: `BDEPEND` in `EAPI-7`.

File: dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+S=${WORKDIR}/${PN}-${MY_PV}
+
+DOCS=( README README.ORG CHANGES INSTALL RELNOTES )
```
Comment (a17r) [inline dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild]: I doubt we really need `INSTALL` here, after it is already installed.

**Comment by mm1ke** [comment]
@a17r 
Fixed, thanks for the review.

File: dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+DOCS=( README README.ORG CHANGES RELNOTES )
+HTML_DOCS=( Jam.html Jambase.html Jamfile.html )
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-2.5.3-nostrip.patch
+	"${FILESDIR}"/${PN}-2.5.3-i-hate-yacc.patch
+	"${FILESDIR}"/${PN}-2.5.3-false-flags.patch
+)
+
+src_prepare() {
```
Comment (SoapGentoo) [inline dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild]: does this use an Autoconf `./configure` script? If no, I suggest changing this to `src_configure` and dropping the `default`, makes things even simpler

Comment (mm1ke) [reply dev-util/ftjam/ftjam-2.5.3_rc2-r2.ebuild]: Actually it is a Autoconf script. I even tried to do what you suggested, but that way it wouldn't build anything.

**Pull request closed without merging**